### Investigating the discharge summary

In [37]:
import pandas as pd
import numpy as np
import os
import psycopg2
import sqlalchemy
import string

In [17]:
# connect to the mimic database and set the search path to the 'mimiciii' schema

dbschema='mimiciii'
cnx = create_engine('postgresql+psycopg2://aa5118:mimic@localhost:5432/mimic',
                    connect_args={'options': '-csearch_path={}'.format(dbschema)})


In [18]:
cat = "'Discharge summary'"
df_temp = pd.read_sql_query('''
  SELECT hadm_id FROM noteevents WHERE category = ''' + cat + '''
''', cnx)
df_temp.shape

(59652, 1)

59652 Discharge summaries from ~2m notes

In [19]:
df_temp = pd.read_sql_query('''
  SELECT hadm_id FROM noteevents WHERE category = ''' + cat + ''' AND hadm_id IS NOT NULL
  ORDER BY hadm_id
''', cnx)
df_temp.shape

(59652, 1)

These all contain a hospital admission ID (`hadm_id`) - there are no NULLs. Only patients who get admitted to the ICU during their hospital admission will have an hadm_id in this table according to https://github.com/MIT-LCP/mimic-code/issues/237. So this means the discharge summary is only for being discharged from the ICU, not the hospital.

In [21]:
df_temp = pd.read_sql_query('''
  SELECT COUNT(DISTINCT hadm_id) FROM noteevents WHERE category = ''' + blah + ''' AND hadm_id IS NOT NULL
  --ORDER BY hadm_id
''', cnx)
df_temp

count
0  52726

However, there are only 52726 unique `hadm_id`s in the table. This shows that patients can have multiple distinct ICU stays during a single hospital admission. Indeed there are ~7k duplicate `hadm_id`s. And for each ICU stay, they will get a discharge summary. This clears the picture a litte with deciding how we can utilise the predictive power of the discharge summaries

In [22]:
# breakdown of note categories showing the number of notes, average number of characters
# and the number of notes in each category where the note time was provided

df_summary = pd.read_sql_query('''
  SELECT 
      category,
      COUNT(category),
      ROUND(AVG(LENGTH(text)),-1) AS text_avg_chars,
      COUNT(charttime) AS time_provided
  FROM noteevents
  GROUP BY category
''', cnx)
df_summary

category   count  text_avg_chars  time_provided
0    Case Management      967          1120.0            967
1             Consult      98          6040.0             98
2   Discharge summary   59652          9620.0              0
3                 ECG  209051           210.0              0
4                Echo   45794          2320.0              0
5             General    8301          1560.0           8260
6             Nursing  223556          1790.0         222172
7       Nursing/other  822497           800.0         822497
8           Nutrition    9418          2430.0           9411
9            Pharmacy     103          2580.0            102
10         Physician   141624          7140.0         141048
11          Radiology  522279          1740.0         522279
12     Rehab Services    5431          3120.0           5429
13       Respiratory    31739          1360.0          31703
14        Social Work    2670          2160.0           2648

Discharge summaries are the longest note category in the `NOTEEVENTS` table with almost 10k characters.

In [23]:
# confirming that the dataframe output should have 0 rows

df_temp = pd.read_sql_query('''
  SELECT charttime FROM noteevents WHERE category = ''' + cat + ''' AND charttime IS NOT NULL
''', cnx)
df_temp

Empty DataFrame
Columns: [charttime]
Index: []

Furthermore, they are one of the few note categories which do not provide a timestamp

#### Cause of death

Below query matches the string "cause of death" in the text field and returns the substring occuring after it until the end of the note

In [55]:
sql = """
SELECT 
    SUBSTRING (n.text FROM 
        POSITION('cause of death' IN n.text) FOR (LENGTH(n.text) - POSITION('cause of death' IN n.text)))
FROM noteevents n
INNER JOIN admissions a
ON n.hadm_id = a.hadm_id
WHERE a.hospital_expire_flag = 1
AND lower(n.text) LIKE '%cause of death%'
AND category LIKE 'Discharge summary%';
"""

df_temp = pd.read_sql_query(sqlalchemy.text(sql), cnx)
df_temp.head()

substring
0  cause of death was hyperkalemia from acute ren...
1  cause of death was likely cardiac arrest.\nSec...
2  cause of death was cardiac arrest secondary to...
3  cause of death being cardiopulmonary\nfailure ...
4  Admission Date:  [**2131-9-2**]              D...

In [56]:
print([word for word in df_temp.iloc[1]])

['cause of death was likely cardiac arrest.\nSecondary cause sepsis.\n\n\n\n\n                          [**Name6 (MD) **] [**Name8 (MD) **], M.D.  [**MD Number(1) 968**]\n\nDictated By:[**Last Name (NamePattern1) 2584**]\n\nMEDQUIST36\n\nD:  [**2140-6-9**]  04:23\nT:  [**2140-6-12**]  16:16\nJOB#:  [**Job Number 2585**]']


Maybe we can use the final discharge summary from an ICU visit to predict whether the patient will die in the ICU at their next visit.